tools/preflight/preflight.go

```diff
@@ -0,0 +1,729 @@
+	}
+
+	op.Logger.Infoln("Checking for required kubernetes server version (>=", minK8sVersion, ")")
+	err = op.checkKubernetesVersion()
+	if err != nil {
+		op.Logger.Errorln(string(cross), err.Error())
+		op.Logger.Errorln(string(cross), "Preflight check for kubernetes version failed")
+		preflightStatus = false
+	} else {
+		op.Logger.Infoln(string(check), "Preflight check for kubernetes version is successful")
+	}
+
+	op.Logger.Infoln("Checking Kubernetes RBAC")
+	err = op.checkKubernetesRBAC()
+	if err != nil {
+		op.Logger.Errorln(string(cross), err.Error())
+		op.Logger.Errorln(string(cross), "Preflight check for kubernetes RBAC failed")
+		preflightStatus = false
+	} else {
+		op.Logger.Infoln(string(check), "Preflight check for kubernetes RBAC is successful")
+	}
+
+	//  Check storage snapshot class
+	op.Logger.Infoln("Checking if a StorageClass and VolumeSnapshotClass are present")
+	err = op.checkStorageSnapshotClass()
+	if err != nil {
+		op.Logger.Errorln(err.Error())
+		op.Logger.Errorln(string(cross), "Preflight check for SnapshotClass failed")
+		storageSnapshotSuccess = false
+		preflightStatus = false
+	} else {
+		op.Logger.Infoln(string(check), "Preflight check for SnapshotClass is successful")
+	}
+
+	//  Check CSI installation
+	op.Logger.Infoln("Checking if CSI APIs are installed in the cluster")
+	err = op.checkCSI()
+	if err != nil {
+		op.Logger.Errorln(err.Error())
+		op.Logger.Errorln("Preflight check for CSI failed")
+		preflightStatus = false
+	} else {
+		op.Logger.Infoln(string(check), "Preflight check for CSI is successful")
+	}
+
+	//  Check DNS resolution
+	op.Logger.Infoln("Checking if DNS resolution is working in k8s cluster")
+	err = op.checkDNSResolution()
+	if err != nil {
+		op.Logger.Errorln(string(cross), err.Error())
+		op.Logger.Errorln(string(cross), "Preflight check for DNS resolution failed")
+		preflightStatus = false
+	} else {
+		op.Logger.Infoln(string(check), "Preflight check for DNS resolution is successful")
+
+	}
+
+	//  Check volume snapshot and restore
+	if storageSnapshotSuccess {
+		op.Logger.Infoln("Checking if volume snapshot and restore is enabled in cluster")
+		err = op.checkVolumeSnapshot()
+		if err != nil {
+			op.Logger.Errorln(string(cross), err.Error())
+			op.Logger.Errorln(string(cross), "Preflight check for volume snapshot and restore failed")
+			preflightStatus = false
+		} else {
+			op.Logger.Infoln(string(check), "Preflight check for volume snapshot and restore is successful")
+		}
+	}
+
+	if !preflightStatus {
+		op.Logger.Warnln("Some preflight checks failed")
+		if op.PerformCleanupOnFail {
+			op.CleanupByID(resNameSuffix)
+		}
+	} else {
+		op.Logger.Infoln("All preflight checks succeeded!")
+		op.CleanupByID(resNameSuffix)
+	}
+}
+
+// checkKubectl checks whether kubectl utility is installed.
+func (o *Options) checkKubectl() error {
+	cmdOut, err := shell.ExecuteArgs([]string{}, true, "bash", "-c", "command -v \"kubectl\"")
```

Comment:
I suppose this command won't run other OS such as windows as it will work on bash/sh shell.
"bash" and "-c" is also required to tell the shell on which binary/program to execute the shell. Otherwise it will look for "command" binary in $PATH of the system.

For other OS support some enhacement  in design will be required. It will require efforts.

Reply:
As discussed, used LookPath function of exec package of go library.